sources/packages/backend/apps/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -240,6 +262,77 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
     });
   }
 
+  /**
+   * Soft deletes a provincial restriction from Student.
+   * @param studentId ID of the student to get a restriction.
+   * @param studentRestrictionId ID of the student restriction to be deleted.
+   * @param auditUserId audit user ID who is deleting the restriction.
+   * @param noteDescription student notes added during deletion of restriction.
+   */
+  async deleteProvincialRestriction(
+    studentId: number,
+    studentRestrictionId: number,
+    auditUserId: number,
+    noteDescription: string,
+  ): Promise<void> {
+    const restriction = await this.repo.findOne({
+      select: {
+        id: true,
+        deletedAt: true,
+      },
+      relations: {
+        restriction: true,
+      },
+      where: {
+        id: studentRestrictionId,
+        student: { id: studentId },
+        restriction: { restrictionType: RestrictionType.Provincial },
+      },
+      withDeleted: true,
+    });
+    if (!restriction) {
+      throw new CustomNamedError(
+        "Provincial restriction not found.",
+        RESTRICTION_NOT_FOUND,
+      );
+    }
+    if (restriction.deletedAt) {
+      throw new CustomNamedError(
+        "Provincial restriction is already set as deleted.",
+        RESTRICTION_IS_DELETED,
+      );
+    }
+    await this.dataSource.transaction(async (transactionalEntityManager) => {
+      const now = new Date();
+      const auditUser = { id: auditUserId } as User;
+      const note = await this.noteSharedService.createStudentNote(
+        studentId,
+        NoteType.Restriction,
+        noteDescription,
+        auditUserId,
+        transactionalEntityManager,
+      );
+      const updateResult = await transactionalEntityManager
+        .getRepository(StudentRestriction)
+        .update(
+          { id: restriction.id, deletedAt: IsNull() },
+          {
+            deletionNote: note,
+            deletedAt: now,
+            deletedBy: auditUser,
+            modifier: auditUser,
+            updatedAt: now,
+          },
+        );
```

Comment:
There was a discussion about setting isActive to false while deleting a restriction(to reduce legacy integration side effects). Is it dropped to keep the deletion process simple?